### DISCUSSION TITLE
Beta Question - Adding Tracks to Playlists

### DISCUSSION THREAD

**Discussion by pjdreger3** [music-assistant/support#1219] [Q&A] [answered]
I can create playlists see them in the playlist folder but when I add tracks to the playlist they don't get added. 

Fairly certain I am running the beta so this just may be a not done yet problem.

**Comment by OzGav** [comment]
No this should work. Look in the logs for any errors. You can open the playlist file with a text editor and see if the tracks were added.

**Comment by pjdreger3** [comment]
Its not working for me. 

Here is where I select the track to add to a playlist
![save to Playlist](https://user-images.githubusercontent.com/7957767/235726752-0d55571a-dbe9-4d58-b763-ba74b7533624.PNG)

here is where the playlists are listed
![list of playlist](https://user-images.githubusercontent.com/7957767/235726936-4d8131dc-ec65-4eeb-992c-6e929ec2e081.PNG)

here is where I look in the playlist after the add
![playlist contents](https://user-images.githubusercontent.com/7957767/235727074-7be72bdb-bc17-4970-86a4-2f267255bea4.PNG)

Checking the logs to see if there are any message
![logs after add to playlist](https://user-images.githubusercontent.com/7957767/235727213-ec5c0d8b-268c-4e51-b6dd-1b3418fb5455.PNG)


I also tried adding the track to the data base and then reloaded and synced the music source.  No luck

I wasn't able to check the data bases . I checked in the FAQ and didnt get any pointers to the location .

**Reply by OzGav** [reply]
You aren’t looking in the MA log. That is the HA core log. On that page look top right and you will see a drop down. Select that and find Music Assistant.

**Comment by pjdreger3** [comment]
first image is wrong. Here's the correct image
![select track](https://user-images.githubusercontent.com/7957767/235728119-7fed2fd5-1ffb-4287-b474-3ce4d2c18859.PNG)

**Comment by pjdreger3** [comment]
So I looked in the log and I am getting this error:
TypeError: decode() argument 'encoding' must be str, not None
2023-05-03 09:52:06 WARNING music_assistant.providers.filesystem_local  -- Error while parsing playlist Spring.m3u
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/providers/filesystem_local/base.py", line 454, in get_playlist_tracks
    playlist_data = playlist_data.decode(encoding_details["encoding"])
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: decode() argument 'encoding' must be str, not None

I opened the track on mp3tag and I don't see a encoding tag.

**Reply by OzGav** [reply]
That doesn't looking like a tagging problem. What is in the playlist file? Can you try deleting it and then try again?

**Comment by pjdreger3** [comment]
Deleted playlist - recreated playlist (it shows in playlist section) . Added track to new playlist - then opened the playlist - track not shown. 

Here are the log entries 
TypeError: decode() argument 'encoding' must be str, not None
2023-05-03 10:17:05 ERROR asyncio  -- Task exception was never retrieved
future: <Task finished name='Task-32676' coro=<PlaylistController.add_playlist_track() done, defined at /usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/media/playlists.py:120> exception=TypeError("decode() argument 'encoding' must be str, not None") created at /usr/local/lib/python3.11/site-packages/music_assistant/server/server.py:254>
source_traceback: Object created at (most recent call last):
  File "/usr/local/bin/mass", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.11/site-packages/music_assistant/__main__.py", line 108, in main
    run(
  File "/usr/local/lib/python3.11/site-packages/aiorun.py", line 273, in run
    loop.run_forever()
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 607, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1914, in _run_once
    handle._run()
  File "/usr/local/lib/python3.11/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/providers/websocket_api/__init__.py", line 244, in _run_handler
    result = await result
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/media/playlists.py", line 118, in add_playlist_tracks
    self.mass.create_task(self.add_playlist_track(db_id, uri))
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/server.py", line 254, in create_task
    task = self.loop.create_task(target)
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/media/playlists.py", line 178, in add_playlist_track
    await provider.add_playlist_tracks(playlist_prov.item_id, [track_id_to_add])
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/providers/filesystem_local/base.py", line 495, in add_playlist_tracks
    playlist_data = playlist_data.decode(encoding_details["encoding"])
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: decode() argument 'encoding' must be str, not None
2023-05-03 10:17:09 WARNING music_assistant.providers.filesystem_local  -- Error while parsing playlist Spring.m3u
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/providers/filesystem_local/base.py", line 454, in get_playlist_tracks
    playlist_data = playlist_data.decode(encoding_details["encoding"])
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: decode() argument 'encoding' must be str, not None

**Reply by OzGav** [reply]
Are you able to add any tracks to any playlists?

**Comment by pjdreger3** [comment]
also - when I return to the MA tab the entire page is covered in blue 
![image](https://user-images.githubusercontent.com/7957767/235944265-a88c8fd0-d149-43f0-8834-8b74a8f860ef.png)

If I refresh the page it returns to normal

**Reply by pjdreger3** [reply]
I have one (1) of five (5) play list I can add tracks to. I created that playlist in the non Beta version of MA.

**Reply by OzGav** [reply]
Just a follow up. There is a bug related to this so keep an eye out for future updates where this will likely be fixed. This is the issue https://github.com/music-assistant/hass-music-assistant/issues/1300 with a possible workaround...

**Reply by OzGav** [reply, marked as answer]
This has been fixed in the latest beta

**Comment by pjdreger3** [comment]
Just tried it out and it works. But, the delete item from playlist does not work.

**Reply by OzGav** [reply]
In my testing it appears that delete from playlist is deleting the following track. If there is only one track in the playlist then the track is not removed. Is that what you are seeing?

**Reply by pjdreger3** [reply]
in my case it wont delete the last item in the playlist. Which seems consistent to your experience of it deleting current position +1

**Reply by OzGav** [reply]
I’ll add it as a bug to be fixed. Thanks for the follow up

**Comment by muttleydosomething** [comment]
I can’t see any way of adding music to a playlist. I’ve spent hours and although I can create a playlist I can’t see any way of adding music to it. Does anyone know how this can be done? Many thanks

![IMG_4944](https://github.com/user-attachments/assets/4a601a52-e4da-470d-85ab-a1f6a94f7a20)

**Reply by muttleydosomething** [reply]
PS If I try “Add to library” I get the error Field “provider_mappings” 

![IMG_4945](https://github.com/user-attachments/assets/284e76ee-4788-4090-9e3f-9bcbc343c30d)

**Reply by OzGav** [reply]
You are commenting on a closed issue from 18 months ago….

**Reply by muttleydosomething** [reply]
Wow, I found it. I think rather than being able to select a folder or multiple files, one needs to go into each track and then add them individually. Whilst this works it can become rather tiresome if one has a large amount of files to add. Still, problem solved 👍🏻

![IMG_4947](https://github.com/user-attachments/assets/78339dc9-f1c2-4200-b9fd-a3e4bcd37e63)

**Reply by muttleydosomething** [reply]
> You are commenting on a closed issue from 18 months ago….

Ah, sorry OzGav. Congrats on Music Assistant Server though.
